### ISSUE TITLE
Prefer GOVERSION to go.mod in DetectGoVersion().

### ISSUE THREAD

**Issue by csilvers** [golangci/golangci-lint#4265]
### Your feature request related to a problem? Please describe.

I have a scenario where I want to specify the go version to run golangci-lint against, but for complicated reasons cannot easily set the `--go` flag.  Right now, in the absence of `--go` the code falls back to checking the version in go.mod, and if that fails the version in GOVERSION.  For us, the version in go.mod is not always correct -- which is allowed, as I understand it, since go.mod is supposed to specify a minimum version, not an actual version.

### Describe the solution you'd like.

Have DetectGoVersion to prefer GOVERSION, and fall back to go.mod only if that is not set.

Since go.mod is technically only supposed to provide a minimum version of Go that is supported, using it to detect the actual go version is a bit suspect, so preferring an explicit envvar seems (to me) to be a reasonable move.

Technically this would be a breaking change and require a major version bump, but maybe we could get away without it?

### Describe alternatives you've considered.

I could try to delete my go.mod file before running golangci-lint, and then restore it afterwards, but that is not ideal.  I could also try really hard to make it so we could set the `--go` flag after all, but that would require a lot of work.  I would do it if, semantically, go.mod were superior to using the envvar, but I don't think that's the csae.

**Comment by ldez** [comment]
Hello,

You are in an edge case, the current algo is the best for the majority of the cases.

You can set the go version inside the configuration file:
```yml
run:
  go: '1.19'
```

https://golangci-lint.run/usage/configuration/#run-configuration
